### PR TITLE
Programmatic PDF form filling - iOS

### PR DESCRIPTION
**Pull request by radazzouz**
# Details

iOS implementation of #70

recording

# Acceptance Criteria

- [x] Add `getFormFieldValue(fullyQualifiedName)` and `setFormFieldValue(value, fullyQualifiedName)` API.
- [x] Add 'Programmatic Form Filling' Catalog example.

### CODE REVIEW

File: ios/RCTPSPDFKit/RCTPSPDFKitView.m [hunk 1]
```diff
@@ -198,6 +197,54 @@ - (void)addAnnotations:(NSString *)jsonAnnotations {
...
+
+  PSPDFDocument *document = self.pdfController.document;
+  for (PSPDFFormElement *formElement in document.formParser.forms) {
+    if ([formElement.fullyQualifiedFieldName isEqualToString:fullyQualifiedName]) {
+      id formFieldValue = formElement.value;
+      return @{@"value": formFieldValue ?: [NSNull new]};
+    }
+  }
+
+  return @{@"error": @"Failied to get the form field value."};
```
Comment (irgendeinich) [inline ios/RCTPSPDFKit/RCTPSPDFKitView.m]: Typo in `Failied` -> `Failed`

File: samples/Catalog/Catalog.ios.js [hunk 2]
```diff
@@ -146,6 +146,17 @@ var examples = [
       component.props.navigator.push(nextRoute)
     },
   },
+  {
+    name: 'Programmatic Form Filling',
+    description:
+      'Shows how to get and add new annotations using Instant JSON.',
```
Comment (irgendeinich) [inline samples/Catalog/Catalog.ios.js]: Description should be updated.

File: ios/RCTPSPDFKit/RCTPSPDFKitView.h [hunk 3]
```diff
@@ -28,10 +28,17 @@
 @property (nonatomic, copy) RCTBubblingEventBlock onAnnotationsChanged;
 @property (nonatomic, copy) RCTBubblingEventBlock onStateChanged;
 
+/// Document
 - (void)saveCurrentDocument;
+
+///Anotations
```
Comment (steviki) [inline ios/RCTPSPDFKit/RCTPSPDFKitView.h]: Misses a space.

File: index.js [hunk 4]
```diff
@@ -219,6 +219,32 @@ class PSPDFKitView extends React.Component {
...
         }
     }
+    
+    /**
+     * Gets the value of the form element of the fully qualified name.
+     * 
+     * @param fullyQualifiedName The fully qualified name of the form element.
+     *
+     * Returns a promise resolving a dictionary with the following structure:
+     * {'value' : value} or ['error' : Failed to get the form field value.]
```
Comment (steviki) [inline index.js]: `['error' : Failed to get the form field value.]` is still shown as an array literal, while it should be an object.

**Pull request merged**